pkg/buffer/bucket.go

```diff
@@ -0,0 +1,114 @@
+package buffer
```

Comment:
Is it possible to put all the `SFU` related stuff in `pkg/sfu` and make sub-directories there? I think it would allow us to keep things clean. Or maybe it is not necessary as I see some code which looks it might be from `ion-sfu` already duplicated in `livekit-server`. Will defer to @davidzhao for advice.